### PR TITLE
Document workload cluster naming considerations and restrictions

### PR DESCRIPTION
**Pull request by AndiDog**
### What does this PR do?

https://github.com/giantswarm/giantswarm/issues/28900

I didn't link to our Kyverno policy since that's pretty much an implementation detail. Let me know if you think it should be referenced.

### Have you maintained the front matter?

yes

### CODE REVIEW

File: src/content/getting-started/create-workload-cluster/index.md [hunk 1]
```diff
@@ -48,9 +48,19 @@ kubectl config use-context gs-wombat
...
+First, please choose a name for the new workload cluster. We recommend you choose a naming scheme suiting your organization, and then stick to it.
 
-First, template a cluster ([command reference]({{< relref "/use-the-api/kubectl-gs/template-cluster" >}})):
+- The maximum cluster name length is 20 characters.
+- Letters, digits and dashes are allowed. Dashes are not possible at beginning or end.
+- For example, `dev-eu-central-1-a`, `dev-a`, `prod-b`, `ecommerce-dev-eu`, `prod-us` or `staging01` are all valid names.
+- The cluster name will be part of domain names such as `api.<cluster name>.<base domain>` (for instance: `api.mycluster.eu-west-1.aws.gigantic.io` or `ingress.mycluster.mycompany.com`). Since DNS records can normally be resolved publically by anyone, you should avoid encoding sensitive information into the cluster name (`mycluster` in our examples).
+- Clusters can't be renamed after creation.
+- Within one management cluster, all workload cluster names must be unique, even across namespaces. This is so that tooling and operators can look up workload clusters by their unique name without the chance of confusion.
```
Comment (fiunchinho) [inline src/content/getting-started/create-workload-cluster/index.md:58]: Do we require different management clusters to use different base domains? Unless we do, clusters can't use the same name even on different management clusters, or DNS names will collide.

File: src/content/getting-started/create-workload-cluster/index.md [hunk 2]
```diff
@@ -48,9 +48,19 @@ kubectl config use-context gs-wombat
...
+- We recommend specifying a cluster name explicitly using the `--name` parameter, as in the below instructions. If you really want a randomly-generated name, you can instead use `--generate-name`.
+
+You will now create resources with `kubectl gs`. In particular, this tutorial uses the `kubectl gs template` command to create valid YAML for each resource. The templating commands do not immediately create the cluster – the resulting YAML manifest must be applied in order to create the cluster. Alternatively, the Web UI provides a visual way to create clusters.
```
Comment (pipo02mix) [inline src/content/getting-started/create-workload-cluster/index.md:61]:
```suggestion
You will now create resources with `kubectl gs`. In particular, this tutorial uses the `kubectl gs template` command to create valid YAML for each resource. The templating commands do not immediately create the cluster – the resulting YAML manifest must be applied to the management cluster API  in order to create the cluster. Alternatively, the Web UI provides a visual way to create clusters.
```

File: src/content/getting-started/create-workload-cluster/index.md [hunk 3]
```diff
@@ -66,6 +76,8 @@ kubectl gs template cluster \
   > cluster.yaml
 ```
 
+For backward compatibility, vintage cluster templating does not require the `--name` parameter. If no name is specified, a random one like `zekfaewnxh` will be generated. The `--name` parameter becomes required with CAPI clusters, but you can also use `--generate-name` if a random name is really desired.
```
Comment (pipo02mix) [inline src/content/getting-started/create-workload-cluster/index.md:79]: Why we introduce `--generate-name` if the default behaviour is that? do we plan to require `name`?

File: src/content/getting-started/create-workload-cluster/index.md (hunk 1 above)
Comment (AndiDog) [reply src/content/getting-started/create-workload-cluster/index.md]: You're right. I extended the description more and gave a strong, general recommendation for uniqueness with more explanation.

File: src/content/getting-started/create-workload-cluster/index.md (hunk 3 above)
Comment (AndiDog) [reply src/content/getting-started/create-workload-cluster/index.md:85]: `--name` is now required for CAPI (see https://github.com/giantswarm/roadmap/issues/2983). For vintage, we remain compatible with old behavior i.e. don't require any of the two parameters and default to random names.

File: src/content/getting-started/create-workload-cluster/index.md [hunk 4]
```diff
@@ -46,11 +46,27 @@ kubectl config use-context gs-MC_NAME
...
+    - Using identical workload cluster names creates problems and risk. Imagine asking for support, or trying to resolve an incident, and Giant Swarm's or your team members confuse two clusters of the same name. Hence, cluster names should be unique, easily spellable and pronounceable by humans – for example during incident calls and in chat – and follow a somewhat consistent naming scheme.
+- We strongly recommend specifying a cluster name explicitly using the `--name` parameter, as in the below instructions. If you really want a randomly-generated name, you can instead use `--generate-name`.
+
+## Step 3: Template and create the workload cluster
+
+You will now create resources with `kubectl gs`. In particular, this tutorial uses the `kubectl gs template` command to create valid YAML for each resource. The templating commands do not immediately create the cluster – the resulting YAML manifest must be applied to the management cluster API  in order to create the cluster. Alternatively, the Web UI provides a visual way to create clusters.
```
Comment (vxav) [inline src/content/getting-started/create-workload-cluster/index.md:67]:
```suggestion
You will now create resources with `kubectl gs`. In particular, this tutorial uses the `kubectl gs template` command to create valid YAML for each resource. The templating commands do not immediately create the cluster – the resulting YAML manifest must be applied to the management cluster API or committed to the Gitops repository in order to create the cluster. Alternatively, the Web UI provides a visual way to create clusters.
```

Comment (vxav) [reply src/content/getting-started/create-workload-cluster/index.md:67]: Same thing in the end but some customers might not see the difference

**Review by vxav — APPROVED** [review]
Nice

**Review by pipo02mix — APPROVED** [review]
LGTM if Jose is happy

**Pull request merged**